rust_snuba/rust_arroyo/src/backends/kafka/mod.rs

```diff
@@ -71,42 +71,32 @@ fn create_kafka_message(msg: BorrowedMessage) -> BrokerMessage<KafkaPayload> {
     )
 }
 
-struct OffsetStage<C: AssignmentCallbacks> {
-    staged_offsets: HashMap<Partition, u64>,
-    consumer: Arc<Mutex<Option<BaseConsumer<CustomContext<C>>>>>,
-}
-
-impl<C: AssignmentCallbacks> CommitOffsets for OffsetStage<C> {
-    fn commit(
-        mut self,
-        offsets: HashMap<Partition, u64>,
-    ) -> Result<HashMap<Partition, u64>, ConsumerError> {
-        self.staged_offsets.extend(offsets);
-        let mut partitions = TopicPartitionList::with_capacity(self.staged_offsets.len());
-        for (partition, offset) in &self.staged_offsets {
+impl<C: AssignmentCallbacks> CommitOffsets for Arc<RwLock<Option<BaseConsumer<CustomContext<C>>>>> {
```

Comment:
can you keep the struct around it? i understand there's nothing wrong with this code but i think it makes it harder to understand for people who are new to rust